initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/container/dockercompose/DockerComposeService.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.spring.container.dockercompose;
+
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+/**
+ * A Docker Compose service.
+ *
+ * @author Moritz Halbritter
+ */
+public class DockerComposeService {
+
+	private static final String INDENT = "  ";
+
+	private final String name;
+
+	private final String image;
+
+	private final String imageTag;
+
+	private final String imageWebsite;
+
+	private final Map<String, String> environment;
+
+	private final List<Integer> ports;
+
+	/**
+	 * Creates a new docker compose service.
+	 * @param name the name of the service
+	 * @param image the image of the service
+	 * @param imageTag the image tag of the service
+	 * @param imageWebsite the image website of the service
+	 * @param environment the environment of the service
+	 * @param ports the ports of the service
+	 */
+	public DockerComposeService(String name, String image, String imageTag, String imageWebsite,
```

Comment:
This should have a builder like the other concepts, rather than a constructor with so many argument. You could also infer nice default where the builder is created with an imageTag and it sets a default name automatically, ports that take a vararg and, why not, a builder static method that takes a `imageTag:version` format.